ledger-core/virtual/virtual.go

```diff
@@ -140,6 +144,11 @@ func (lr *Dispatcher) Init(ctx context.Context) error {
 	lr.Conveyor.AddInterfaceDependency(&lr.messageSenderAdapter)
 	lr.Conveyor.AddInterfaceDependency(&lr.memoryCacheAdapter)
 	lr.Conveyor.AddInterfaceDependency(&lr.AuthenticationService)
+	lr.Conveyor.AddInterfaceDependency(&lr.ReferenceBuilder)
```

Comment:
better to not have it as a dependency - it should be created locally by using the scheme
behavior customization should be via the cryptography scheme pls

Reply:
will fix in next patch